SQL/SQL.sublime-syntax

```diff
@@ -243,3 +249,31 @@ contexts:
       # Handle situations where the schema and . 
     - match: '{{end_identifier}}'
       pop: true
+  like_escape:
+    - match: (\')([^'])(\')
+      captures:
+        0: meta.string.escape.sql string.quoted.single.sql
+        1: punctuation.definition.string.begin.sql
+        2: constant.character.escape.sql
+        3: punctuation.definition.string.end.sql
+      pop: true
+    - match: (?=\S)
+      pop: true
+  like_string:
+    - match: \'
+      scope: punctuation.definition.string.begin.sql
+      set: inside_like_single_quoted_string
+    - match: (?=\S)
+      pop: true
+  inside_like_single_quoted_string:
+    - meta_scope: meta.string.like.sql string.quoted.single.sql
+    - match: \'
+      scope: punctuation.definition.string.end.sql
+      pop: true
+    - match: (\[)(?:.|[^]']+?)(\])
```

Comment:
It seems, sets behave pretty much like regexp character groups. Do we want to honor it?

Reply:
They do indeed, though much more limited as I understand. It seems it is just an optional negation operator `^` (which I need to update this PR to scope correctly) followed by a set of characters. Nothing fancy with `-` or `&` etc.
https://codingsight.com/t-sql-regular-expression-like-operator-and-its-use-cases/

Reply:
It would be nice to be able to just re-use the Regex syntax def for this, but I guess we'd need to wait for https://github.com/sublimehq/Packages/issues/1942#issuecomment-905777773 first.

Reply:
There are so many things to doooo....